rundeckapp/src/main/groovy/rundeck/services/ExecutionLifecycleComponentService.groovy

```diff
@@ -18,28 +17,56 @@ import com.dtolabs.rundeck.core.plugins.SimplePluginConfiguration
 import com.dtolabs.rundeck.plugins.ServiceNameConstants
 import com.dtolabs.rundeck.plugins.jobs.ExecutionLifecyclePlugin
 import com.dtolabs.rundeck.plugins.jobs.JobExecutionEventImpl
-import com.dtolabs.rundeck.plugins.project.JobLifecyclePlugin
 import com.dtolabs.rundeck.server.plugins.services.ExecutionLifecyclePluginProviderService
+import groovy.transform.CompileDynamic
 import groovy.transform.CompileStatic
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.ApplicationContext
+import org.springframework.context.ApplicationContextAware
 import rundeck.ScheduledExecution
+import rundeck.services.feature.FeatureService
 
 /**
  * Provides capability to execute certain task based on a job execution event
  * Created by rnavarro
  * Date: 5/07/19
  * Time: 10:32 AM
  */
-class ExecutionLifecycleComponentService implements IExecutionLifecycleComponentService {
+@CompileStatic
+class ExecutionLifecycleComponentService implements IExecutionLifecycleComponentService, ApplicationContextAware  {
 
-    PluginService pluginService
+    public static Logger log = LoggerFactory.getLogger(ExecutionLifecycleComponentService.class.name)
```

Comment:
you can apply `@Slf4j` annotation to the class to set up the logger automatically